File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size()-1;
        int p2 = n, p1, a;
        p1 = a = 0;
        while(p1<p2){
            a = max(a,n*min(height[p1],height[p2]));
            if(height[p1]<height[p2]) p1++;
            else p2--;
            n--;
        }
        return a;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        mostWater = 0
        while(left != right):
            mostWater = max(mostWater,(right - left) * min(height[left],height[right]))
            if ( height[left] < height[right]):
                left += 1
            else:
                right -= 1
        return mostWater

